File: Documentation/API/Parsers/Midi/ReadNotesFromMidiData.js
import { readFileSync } from 'node:fs';

import RhythmGameUtilities from '@rhythm-game-utilities/core';

const data = readFileSync('./tests/Mocks/song.mid');

const notes = RhythmGameUtilities.ReadNotesFromMidiData(data);

for (let note of notes) {
  console.log(`${note.position} ${note.handPosition}`);
}

File: Documentation/API/Parsers/Midi/ReadNotesFromMidiData.lua
---@type RhythmGameUtilities
local rhythmgameutilities = require("rhythmgameutilities")

local file = io.open("./tests/Mocks/song.mid", "rb")

if file then
    local data = file:read("*a")

    local notes = rhythmgameutilities.read_notes_from_midi_data(data);

    for _, note in pairs(notes) do
        print(note["position"] .. " " .. note["hand_position"])
    end
end
